experimental/client-csharp/src/Wandb/Wandb.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Wandb.Internal;
+
+namespace Wandb
+{
+    public class Session : IDisposable
+    {
+        private readonly Manager _manager;
+        private bool _isInitialized;
+        private int? _port;
+
+        public Session()
+        {
+            _manager = new Manager();
+        }
+
+        public async Task Setup()
+        {
+            // TODO: move this logic to manager
+            if (_isInitialized)
+            {
+                return;
+            }
+            // TODO: get timeout from settings
+            var timeout = TimeSpan.FromSeconds(30);
+            _port = await _manager.LaunchCore(timeout).ConfigureAwait(false);
+            _isInitialized = true;
+        }
+
+        public async Task<Run> Init(

```

Comment:
Does Init takes care of restart scenario? If same experiment name is passed?

Reply:
Yes. Please see the example: https://github.com/wandb/wandb/pull/8286/files#diff-5d92c668d72a035fe8927859e8e81031835a70ca0e4f2f388099186fd994e878R35